docs/ebpf/concepts/loader.md

```diff
@@ -0,0 +1,92 @@
+# Loading Objects
+
+{{ proj }} ships an eBPF object (ELF) loader that aims to be compatible with the
+upstream libbpf and iproute2 (`tc`/`ip`) projects. An ELF is typically obtained
+by compiling a eBPF C program using the LLVM toolchain (`clang`).
+
+This page describes the journey from compiled eBPF ELF to resources in the
+kernel. This involves parsing the ELF into intermediate Go (Spec) types that
+can be modified and copied before loading them into the kernel.
+
+```mermaid
+flowchart LR;
+  A(BPF ELF Object\n\nCompiler output) ==LoadCollectionSpec==> B("CollectionSpec\n\nMapSpecs\nProgramSpecs\nTypes (BTF)");
```

Comment:
True, the subgraphs do make things a lot clearer. I would make the arrows point the other direction, though. This indicates the flow of data, not the order of dependency. IMO this would make more sense to a user. I also replaced `btf.Spec` with `Types` since that's what the field is called in the CollectionSpec.

Speaking of `ARCHITECTURE.md`, wouldn't it be better to absorb that page into the docs as well? And use it as a jump-off point/hook for documenting things like sys.FD?

Reply:
I think we've discussed this before :D Go for turning it around if it makes sense.

Yes re ARCHITECTURE, but that for me is firmly contributor docs. As I understood you wanted to focus on user documentation first?

Reply:
Setting up the beginnings of contributor docs would be a good follow-up.